packages/tools/webpack-fluid-loader/src/protocol.ts

```diff
@@ -0,0 +1,213 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import { IAudienceOwner, IConnectionDetails } from "@fluidframework/container-definitions";
+import {
+    ConnectionHandler,
+    IProtocolDetails,
+    IProtocolHandler,
+    ProtocolHandlerBuilder,
+} from "@fluidframework/container-loader";
+import { IQuorumSnapshot, IScribeProtocolState } from "@fluidframework/protocol-base";
+import {
+    IQuorum,
+    IDocumentAttributes,
+    ISequencedDocumentMessage,
+    IProcessMessageResult,
+    ISignalMessage,
+    IClient,
+    IQuorumEvents,
+    ISequencedClient,
+    ISignalClient,
+} from "@fluidframework/protocol-definitions";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
+
+class EmptyAudience extends EventEmitter implements IAudienceOwner {
+    getMembers(): Map<string, IClient> {
+        return new Map<string, IClient>();
+    }
+
+    getMember(_clientId: string): IClient | undefined {
+        return undefined;
+    }
+
+    removeMember(_clientId: string): boolean {
+        return true;
+    }
+
+    addMember(_clientId: string, _details: IClient) { }
+    clear() { }
+}
+
+class LocalQuorum extends TypedEventEmitter<IQuorumEvents> implements IQuorum {
+    disposed: boolean = false;
+
+    private readonly proposals = new Map<string, any>();
+    private readonly members: Map<string, ISequencedClient> = new Map<string, ISequencedClient>();
+
+    constructor(quorumSnapshot: IQuorumSnapshot) {
+        super();
+
+        for (const pair of quorumSnapshot.values) {
+            const proposal = pair[1];
+            this.proposals.set(proposal.key, proposal.value);
+        }
+    }
+
+    private broadcastClients() {
+        for (const [member, client] of this.members) {
```

Comment:
why does it broadcast all existing clients?

Reply:
this is an experiment left-over